cxx-checks/src/main/java/org/sonar/cxx/checks/FunctionCognitiveComplexityCheck.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Sonar C++ Plugin (Community)
+ * Copyright (C) 2010-2017 SonarOpenCommunity
+ * http://github.com/SonarOpenCommunity/sonar-cxx
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.cxx.checks;
+
+import org.sonar.check.Priority;
+import org.sonar.check.Rule;
+import org.sonar.check.RuleProperty;
+import org.sonar.cxx.api.CxxMetric;
+import org.sonar.cxx.parser.CxxGrammarImpl;
+import org.sonar.squidbridge.api.SourceFunction;
+import org.sonar.squidbridge.checks.ChecksHelper;
+import org.sonar.squidbridge.checks.SquidCheck;
+import com.sonar.sslr.api.AstNode;
+import com.sonar.sslr.api.Grammar;
+import org.sonar.squidbridge.annotations.ActivatedByDefault;
+import org.sonar.squidbridge.annotations.SqaleLinearWithOffsetRemediation;
+import org.sonar.cxx.tag.Tag;
+
+@Rule(
+  key = "FunctionCognitiveComplexity",
```

Comment:
You should add an description, like: https://github.com/SonarSource/SonarJS/blob/149e36eb12db8af81cb538e09f555639dcb3a691/javascript-checks/src/main/resources/org/sonar/l10n/javascript/rules/javascript/S3776.html

Reply:
Added sonar-cxx-plugin/src/main/resources/org/sonar/l10n/c++/rules/cxx/FunctionCognitiveComplexity.html and sonar-c-plugin/src/main/resources/org/sonar/l10n/c/rules/c/FunctionCognitiveComplexity.html (not sure whether we wanted both but I presume this will work with C as well as C++)